src/platform/silabs/wifi/SiWx/WifiInterfaceImpl.cpp

```diff
@@ -439,7 +439,7 @@ sl_status_t SetWifiConfigurations()
             .security = security,
             .encryption = SL_WIFI_DEFAULT_ENCRYPTION,
             .client_options = SL_WIFI_JOIN_WITH_SCAN,
-            .credential_id = SL_NET_DEFAULT_WIFI_CLIENT_CREDENTIAL_ID,
+            .credential_id = (security == SL_WIFI_OPEN) ? SL_NET_NO_CREDENTIAL_ID : SL_NET_DEFAULT_WIFI_CLIENT_CREDENTIAL_ID,
```

Comment:
The changes look okay, but need clarity on what functionality is provided by `credential_id`, looking at the `enum` I am not getting clarity on the purpose of this configuration.